ui/src/message_popup/mobile_prototype/message_components.jsx

```diff
@@ -0,0 +1,178 @@
+/* @flow weak */
+import React from 'react';
+import ReactDOM from 'react-dom';
+import _ from 'lodash';
+
+import Paper from 'material-ui/Paper';
+import IconButton from 'material-ui/IconButton';
+
+import FaceIcon from 'material-ui/svg-icons/action/face';
+import InfoOutlineIcon from 'material-ui/svg-icons/action/info-outline';
+
+const Message = React.createClass({
+  propTypes: {
+    type: React.PropTypes.string.isRequired,
+    text: React.PropTypes.string.isRequired,
+    messageStyle: React.PropTypes.object.isRequired,
+    messageTextStyle: React.PropTypes.object.isRequired,
+    label: React.PropTypes.string,
+    onOpenDialog: React.PropTypes.func
+  },
+  
+  componentDidMount(){ 
+    ReactDOM.findDOMNode(this).scrollIntoView(); 
+  },
+  
+  render(){
+    const {type, messageStyle, messageTextStyle} = this.props;
+    
+    return (
+      <div style={messageStyle}>
+        { type === 'student' && 
```

Comment:
It seems like the core idea here is that `Message` can have a different icon, or maybe no icon.  Instead of putting the details of the `StudentMessage` and `InfoMessage` inside here, could you update `Message` to take an `icon` and then pass in those values from the particular message components?